src/main/java/edu/kit/kastel/sdq/artemis4j/client/ResultDTO.java

```diff
@@ -23,4 +28,44 @@ public record ResultDTO(@JsonProperty long id, @JsonProperty ZonedDateTime compl
     public static ResultDTO forAssessmentSubmission(long submissionId, double score, List<FeedbackDTO> feedbacks, UserDTO assessor) {
         return new ResultDTO(submissionId, null, true, score, true, feedbacks, assessor, AssessmentType.SEMI_AUTOMATIC);
     }
+
+    private List<FeedbackDTO> fetchFeedbacks(ArtemisClient client, long participationId) throws ArtemisNetworkException {
+        if (this.feedbacks != null) {
+            return Collections.unmodifiableList(this.feedbacks);
+        }
+
+        return Arrays.asList(ArtemisRequest.get().path(List.of("participations", participationId, "results", this.id(), "details")).executeAndDecode(client,
+                FeedbackDTO[].class));
+    }
+
+    /**
+     * This will fetch the feedbacks for this result and load the long feedbacks if
+     * necessary.
+     *
+     * @param client          the client to use
+     * @param participationId the participation id, likely
+     *                        {@link ProgrammingSubmission#getParticipationId()}
+     * @return the feedbacks
+     * @throws ArtemisNetworkException if it fails to fetch the feedbacks
+     */
+    public List<FeedbackDTO> fetchDetailedFeedbacks(ArtemisClient client, long participationId) throws ArtemisNetworkException {
+        // Sometimes the feedbacks are not loaded, to fetch the long feedbacks, we need
+        // to load the feedbacks first
+        List<FeedbackDTO> feedbacks = this.fetchFeedbacks(client, participationId);
```

Comment:
This is the only non-static DTO method that performs a network request. For consistency, I'd prefer this to be a static method. What do you think?

Reply:
I was wondering why they were static, I can change that

Reply:
They are static because the client API is entirely stateless. I don't want a user to construct e.g. a full CourseDTO to fetch a specific exercise - having the ID of both should be enough since that's all that Artemis needs.